setup.md

```diff
@@ -2,6 +2,98 @@
 title: Setup
 ---
 
-TBC
+To complete this course you need a computer with access to the internet. All attendees
+should follow the steps below to install the necessary tools and download the exercises
+*before the session*.
+
+## Installing tools
+
+Firstly, you need to install a C++ compiler if you do not have one already. It should
+support the C++14 standard (which all modern compilers do). You also need
+[CMake](https://cmake.org/), at least version 3.11.
+
+You will need a text editor or IDE to work with the example code. For Linux and macOS
+users, [Visual Studio Code] is a good option, but for Windows users we strongly
+recommend that you use [Visual Studio Community], which includes a compiler, CMake and
+an IDE.
+
+Nowadays, many popular IDEs, including Visual Studio and Visual Studio Code, provide
+built-in support for CMake and CTest and will automatically configure your project if a
+CMakeLists.txt file is discovered in its root. For Visual Studio Code, you will need the
+[CMake extension] installed.
+
+[Visual Studio Code]: https://code.visualstudio.com/
+[Visual Studio Community]: https://visualstudio.microsoft.com/vs/community/
+[CMake extension]: https://marketplace.visualstudio.com/items?itemName=ms-vscode.cmake-tools
+
+### Windows: Installing Visual Studio Community
+
+Windows users can [download Visual Studio Community here]. When installing it, make sure
+you select the relevant options for C++ development.
+
+[download Visual Studio Community here]: https://visualstudio.microsoft.com/vs/community/
+
+### macOS: Installing clang and CMake
+
+If you are developing on macOS, it is easiest to install your developer tools via
+Homebrew. If you do not have Homebrew installed, please [visit the Homebrew website] and
+follow the instructions to install it.
+
+Then you can install [the clang compiler] and CMake like so:
+
+```sh
+brew install llvm cmake
+```
```

Comment:
A couple of notes on this. Homebrew requires Xcode command line tools to be installed which already provides a clang compiler. Also, Homebrew is kind of conservative which how it installs things, so when installing `llvm` it doesn't put it on the PATH. The default clang version is 14 and the [llvm package in Homebrew](https://formulae.brew.sh/formula/llvm) appears to use 16 (I think, not sure if this is relevant).

Some output from when I run things:

```sh
$ clang --version
Apple clang version 14.0.3 (clang-1403.0.22.14.1)
Target: arm64-apple-darwin22.5.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin
```

(Note how below it mentions "llvm is keg-only")
```sh
$ brew install llvm cmake
[Cropped out downloading loading bars]
==> Installing llvm
==> Pouring llvm--16.0.6.arm64_ventura.bottle.tar.gz
==> Caveats
To use the bundled libc++ please add the following LDFLAGS:
  LDFLAGS="-L/opt/homebrew/opt/llvm/lib/c++ -Wl,-rpath,/opt/homebrew/opt/llvm/lib/c++"

llvm is keg-only, which means it was not symlinked into /opt/homebrew,
because macOS already provides this software and installing another version in
parallel can cause all kinds of trouble.

If you need to have llvm first in your PATH, run:
  echo 'export PATH="/opt/homebrew/opt/llvm/bin:$PATH"' >> ~/.zshrc

For compilers to find llvm you may need to set:
  export LDFLAGS="-L/opt/homebrew/opt/llvm/lib"
  export CPPFLAGS="-I/opt/homebrew/opt/llvm/include"
==> Summary
🍺  /opt/homebrew/Cellar/llvm/16.0.6: 6,764 files, 1.5GB
==> Running `brew cleanup llvm`...
Disable this behaviour by setting HOMEBREW_NO_INSTALL_CLEANUP.
Hide these hints with HOMEBREW_NO_ENV_HINTS (see `man brew`).
==> Pouring cmake--3.27.1.arm64_ventura.bottle.tar.gz
==> Caveats
To install the CMake documentation, run:
  brew install cmake-docs

Emacs Lisp files have been installed to:
  /opt/homebrew/share/emacs/site-lisp/cmake
==> Summary
🍺  /opt/homebrew/Cellar/cmake/3.27.1: 3,284 files, 55.4MB
==> Running `brew cleanup cmake`...
==> Caveats
==> llvm
To use the bundled libc++ please add the following LDFLAGS:
  LDFLAGS="-L/opt/homebrew/opt/llvm/lib/c++ -Wl,-rpath,/opt/homebrew/opt/llvm/lib/c++"

llvm is keg-only, which means it was not symlinked into /opt/homebrew,
because macOS already provides this software and installing another version in
parallel can cause all kinds of trouble.

If you need to have llvm first in your PATH, run:
  echo 'export PATH="/opt/homebrew/opt/llvm/bin:$PATH"' >> ~/.zshrc

For compilers to find llvm you may need to set:
  export LDFLAGS="-L/opt/homebrew/opt/llvm/lib"
  export CPPFLAGS="-I/opt/homebrew/opt/llvm/include"
==> cmake
To install the CMake documentation, run:
  brew install cmake-docs
```

Reply:
This makes sense. I've now updated the text to tell users to install the `xcode` tools instead (I copied it from the git course).

Clang v14 should be totally fine for what we're doing. We're only using C++14, so nothing bleeding edge.